File: web/client/src/types/feed.ts
import { OriginArticleType } from "./article";
import { OriginCategoryType } from "./category";
import { OriginMyFeedType } from "./myFeed";
import { OriginPlatformType } from "./platform";

export type OriginFeedType = {
  id: string;
  platformId: string;
  categoryId: string;
  name: string;
  description: string;
  rssUrl: string;
  siteUrl: string;
  thumbnailUrl: string;
  trendPlatformType: number;
  apiQueryParam?: string;
  createdAt: string;
  updatedAt: string;
  deletedAt?: string;
};

export type FeedType = OriginFeedType & {
  category: OriginCategoryType;
  platform: OriginPlatformType;
  articles?: Array<OriginArticleType>;
  myFeeds: Array<OriginMyFeedType>;
  isFollowing?: boolean;
};

File: micro-service/content-service/domain/feed.go
package domain

import "time"

type Feed struct {
	ID                string
	Name              string
	RssURL            string
	TrendPlatformType TrendPlatformType
	Platform          Platform
	Category          Category
	CreatedAt         time.Time
	UpdatedAt         time.Time
	DeletedAt         *time.Time
}
